Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again

Hi!

I tried running `mvn quarkus:dev` (or `quarkus dev`) in chapter 2 folder, but I am getting the following error:
```
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalStateException: Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:138)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: java.lang.IllegalStateException: Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:145)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:97)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:132)
        ... 1 more
Caused by: java.lang.IllegalStateException: Hot deployment of the application is not supported when updating the Quarkus version. The application needs to be stopped and dev mode started up again
        at io.quarkus.deployment.dev.IsolatedDevModeMain.close(IsolatedDevModeMain.java:367)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:512)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:68)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:142)
        ... 3 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

I am using an Arch Linux variant (Garuda) with Java 11.0.15. Any ideas/pointers to solve this?

--- a/chapter-2/code-with-quarkus/pom.xml
+++ b/chapter-2/code-with-quarkus/pom.xml
@@ -15,10 +15,10 @@
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <quarkus-plugin.version>2.8.2.Final</quarkus-plugin.version>
+    <quarkus-plugin.version>2.16.4.Final</quarkus-plugin.version>
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
-    <quarkus.platform.version>2.7.5.Final</quarkus.platform.version>
+    <quarkus.platform.version>2.16.4.Final</quarkus.platform.version>
     <surefire-plugin.version>2.22.2</surefire-plugin.version>
   </properties>
 

--- a/chapter-2/quarkus-hello/pom.xml
+++ b/chapter-2/quarkus-hello/pom.xml
@@ -15,10 +15,10 @@
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <quarkus-plugin.version>2.8.2.Final</quarkus-plugin.version>
+    <quarkus-plugin.version>2.16.4.Final</quarkus-plugin.version>
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
-    <quarkus.platform.version>2.7.5.Final</quarkus.platform.version>
+    <quarkus.platform.version>2.16.4.Final</quarkus.platform.version>
     <surefire-plugin.version>2.22.2</surefire-plugin.version>
   </properties>
 

